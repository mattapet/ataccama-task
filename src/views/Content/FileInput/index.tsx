//#region imports
import React from 'react';
import { Upload } from 'antd';
//#endregion

//#region Component interfaces
export interface Props {
  /**
   * A callback invoked when a file gets selected.
   *
   * @param file Selected file.
   */
  readonly onSelect: (file: File) => any;
}
//#endregion

/**
 * Presents a drag-n-drop input for JSON files.
 */
const FileInput: React.FunctionComponent<Props> = props => (
  <Upload.Dragger
    showUploadList={false}
    accept="application/json"
    customRequest={(params: { file?: File }) =>
      params.file && props.onSelect(params.file)
    }
  >
    <h3>Drop JSON File</h3>
  </Upload.Dragger>
);

export default FileInput;
