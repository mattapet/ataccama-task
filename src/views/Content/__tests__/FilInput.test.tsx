//#region imports
import React from 'react';
import renderer from 'react-test-renderer';

import FileInput from '../FileInput';
//#endregion

it('should render correctly', () => {
  // Arrange
  // Act
  const component1 = renderer.create(<FileInput onSelect={() => {}} />);
  const component2 = renderer.create(<FileInput onSelect={() => {}} />);

  // Assert
  expect(component1).toMatchSnapshot();
  expect(component2).toMatchSnapshot();
});
