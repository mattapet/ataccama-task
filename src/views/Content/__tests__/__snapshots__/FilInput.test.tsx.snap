// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<span
  className=""
>
  <div
    className="ant-upload ant-upload-drag"
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span
      className="ant-upload ant-upload-btn"
      onClick={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": undefined,
        }
      }
      tabIndex="0"
    >
      <input
        accept="application/json"
        directory={null}
        multiple={false}
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
        webkitdirectory={null}
      />
      <div
        className="ant-upload-drag-container"
      >
        <h3>
          Drop JSON File
        </h3>
      </div>
    </span>
  </div>
</span>
`;

exports[`should render correctly 2`] = `
<span
  className=""
>
  <div
    className="ant-upload ant-upload-drag"
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span
      className="ant-upload ant-upload-btn"
      onClick={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": undefined,
        }
      }
      tabIndex="0"
    >
      <input
        accept="application/json"
        directory={null}
        multiple={false}
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
        webkitdirectory={null}
      />
      <div
        className="ant-upload-drag-container"
      >
        <h3>
          Drop JSON File
        </h3>
      </div>
    </span>
  </div>
</span>
`;
